src/tsolvers/lrasolver/LRASolver.cc

```diff
@@ -101,9 +101,11 @@ lbool LRASolver::getPolaritySuggestion(PTRef ptref) const {
 // Compute interpolants for the conflict
 //
 PTRef
-LRASolver::getInterpolant( const ipartitions_t & mask , map<PTRef, icolor_t> *labels, PartitionManager &pmanager) {
+LRASolver::getInterpolant( const ipartitions_t & mask , std::map<PTRef, icolor_t> *labels, PartitionManager &pmanager) {
     assert(status == UNSAT);
-    FarkasInterpolator interpolator(logic, explanation, explanationCoefficients, labels,
+    vec<PtAsgn> explCopy;
+    explanation.copyTo(explCopy);
+    FarkasInterpolator interpolator(logic, std::move(explCopy), explanationCoefficients, labels ? *labels : std::map<PTRef, icolor_t>{},
                                     std::unique_ptr<TermColorInfo>(new GlobalTermColorInfo(pmanager, mask)));
     auto itpAlgorithm = config.getLRAInterpolationAlgorithm();
```

Comment:
I'm curious about this mask situation here.  Should it be implicitly reusable for the user, or should the ownership be transferred to by default to the call through the API?  This might be an API-breaking change and we'd need to update the version number, but at the moment I think it would make the design cleaner.  What do you say?

Reply:
I am not sure I get your question. What should or should not be reusable?

Reply:
The question is related to the (resolved) discussion above.  We agreed to relieve the user of the burden of doing copies and, where necessary, to do them transparently inside opensmt.